fitIn/situationHandler.swift

```diff
@@ -12,35 +12,72 @@ import Foundation
 //events
 //Controls flow of app
 struct situationHandler {
+    // MARK: VARIABLES
     
-    private var user = userProfile() //User Data, Info stored here
-    
+    private var user = userProfile(userID: "insertUserID") //User Data, Info stored here
     // instantiation of situation, only one situation is loaded at a time
     //
     // FUTURE: Maybe preload upcoming situation
-    private var currentSituation = situation()
+    private var currentSituation = situation(situationID: "insertSituationID", type: situation.responseType.yesOrNo(true)) {
+        // This didSet assumes that we have segued to our next situation and we are initializing our handler
+        willSet{
+            imageData = (nextSituation?.getImageData())!
+        }
+        didSet {
+            // Clear previous input answer and upcoming situation
+            voteChoice = nil
+            nextSituation = nil
+        }
+    }
+
+    // preload next situation
+    private var nextSituation : situation? {
+        didSet {
+            nextSituationType = nextSituation?.getSituationType()
+        }
+    }
+    private var nextSituationType : String?
     
     //Image Data to use for UIImageView
     private var imageData = Data()
     
-    //lodge a vote. May be too deep of proSocial calls
-    mutating func voteProSocial() {
-        if currentSituation.isProSocial(){
-            user.gotCorrect()
+    // General container for all response types
+    // Handles segue
+    var voteChoice : situation.responseType? {
+        // When the controller sets voteChoice it will automatically call the vote function
+        // Insert transition methods here
+        didSet {
+            if vote() == nil {
+                print("failedVote")
+            }
         }
-        user.gotIncorrect()
     }
-    mutating func voteAntiSocial() {
-        if !currentSituation.isProSocial(){
+  
+    
+    //MARK: METHODS
+    //lodge a vote.
+    // Pre: voteChoice is set to specific case of responseType with its associatedValue
+    // Post: Returns bool? based on right/wrong answer or a nil bug
+    mutating func vote() -> Bool?{
+        if voteChoice == nil {
+            return nil
+        }
+        currentSituation.inputAnswer = voteChoice
+        if currentSituation.isRightAnswer()!{
             user.gotCorrect()
+            return true
         }
         user.gotIncorrect()
```

Comment:
The user is always incorrect?

Reply:
the return of this function just indicates that a vote was successfully counted, I will change the return of the first if condition to false and the overall return of the function to bool instead of bool?

Reply:
Oh, I missed that this is fine.